### PR TITLE
First batch of cherry-picks to 1.6 branch (#16711, #16737, #16408)

### PR DESCRIPTION
**Pull request by ptrendx**
## Description ##
Cherry-picking #16711, #16737 and #16408 to 1.6 branch.

@pengzhao-intel @ChaiBapchya @eric-haibin-lin @Caenorst FYI

## Comments ##
- #16408 is technically not a bug fix, but merging it because:
    - it was in scope for 1.6 release and ready for the code freeze, but blocked because of misunderstanding
    - it is a contrib op, that does not touch any core functionality of MXNet

### CODE REVIEW

File: tests/python/unittest/test_numpy_op.py [hunk 1]
```diff
@@ -2871,6 +2871,118 @@ def check_svd(UT, L, V, data_np):
                 check_svd(UT, L, V, data_np)
 
 
+@with_seed()
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: This test is not related, please remove it.

Comment (ptrendx) [reply tests/python/unittest/test_numpy_op.py]: :+1:

Comment (ChaiBapchya) [reply tests/python/unittest/test_numpy_op.py]: So while we cherrypick the PR can we just select part of that PR? only thing needed from that PR is the change in `src/operator/nn/softmax-inl.h`

Comment (haojin2) [reply tests/python/unittest/test_numpy_op.py]: @ChaiBapchya Actually we also need the mixed precision true_divide... d2l-en depends on that now.

**Comment by haojin2** [comment]
@ptrendx Please go ahead and merge, I'm not sure whether or not I should squash or not.
